app/js/components/identifiers.jsx

```diff
@@ -10,342 +10,381 @@ import React from 'react';
-          identifiers_array: res.results
-        }))
-      })
-    apiCall(null, 'get', 'patientidentifiertype')
-      .then(res => {
-        this.setState(Object.assign({}, this.state, {
-          identifierstypes_array: res.results,
-        }))
-      })
-    apiCall(null, 'get', 'location')
-      .then(res => {
-        this.setState(Object.assign({}, this.state, {
-          location_array: res.results
-        }))
-      })
-  }
+	constructor(props) {
+		super(props);
+		this.state = {
+			patientUuid: props.uuid,
+			identifierUuid: '',
+			activeCard: '',
+			editState: false,
+			newState: false,
+			editable: false,
+			voided: false,
+			prefferedIdentifierUuid: '',
+			identifier: '',
+			identifierType: '',
+			location: '',
+			identifiersArray: [],
+			identifierstypesArray: [],
+			locationArray: [],
+			addIdentifiers: {
+				identifier: '',
+				identifierType: '',
+				location: '',
+				preferred: false
+			},
+			editIdentifiers: {
+				identifier: '',
+				identifierType: '',
+				location: '',
+				preferred: false,
+			},
+			editErrors: {
+				uuid: '',
+				error: ''
+			},
+			addErrors: {
+				uuid: '',
+				error: ''
+			}
+		};
+		this.callEdit = this.callEdit.bind(this);
+		this.callSave = this.callSave.bind(this);
+		this.handleChange = this.handleChange.bind(this);
+		this.callCancel = this.callCancel.bind(this);
+		this.callNew = this.callNew.bind(this);
+		this.callCreate = this.callCreate.bind(this);
+		this.handleCreateChange = this.handleCreateChange.bind(this);
+		this.handleDelete = this.handleDelete.bind(this);
+		this.handlePreferred = this.handlePreferred.bind(this);
+		this.handleError = this.handleError.bind(this);
+		this.reload = this.reload.bind(this);
+	}
 
-  /**
-   * reload component.
-   */
-  reload() {
-    apiCall(null, 'get', `patient/${this.state.patient_uuid}/identifier?v=full`)
-      .then((res) => {
-        this.setState(Object.assign({}, this.state, {
-          identifiers_array: res.results,
-        }))
-      })
-  }
-  /**
-   *
-   * @param {*} isPreferred
-   * @param {*} identifier_uuid
-   *It handle change of the preferred identifier
-   */
-  handlePreferred(isPreferred, identifier_uuid) {
-    let new_editIdentifiers = Object.assign({}, this.state.editIdentifiers);
-    if (!isPreferred) {
-      new_editIdentifiers.preferred = !isPreferred;
-      this.setState({ editIdentifiers: new_editIdentifiers });
-    }
-  }
-  /**
-   *
-   * @param {*} e
-   *Handles chnage of input during edit
-   */
-  handleChange(e, index) {
-      e.preventDefault();
-        const { name, value } = e.target;
-        this.setState({
-          editIdentifiers: Object.assign({}, this.state.editIdentifiers, {
-            [name]: value
-          })
-        })
-  }
+	componentDidMount() {
+		apiCall(null, 'get',
+        `patient/${this.state.patientUuid}/identifier?v=full`)
+			.then((res) => {
+				this.setState(Object.assign({}, this.state, {
+					identifiersArray: res.results
```

Comment:
The code looks good, but still looks like you forgot formatting your code as referenced in OpenMrs Wiki https://wiki.openmrs.org/display/docs/JavaScript+Conventions and https://github.com/felixge/node-style-guide#2-spaces-for-indentation